src/gtc/python/npir_gen.py

```diff
@@ -230,18 +230,21 @@ def visit_EmptyTemp(
 
     VectorTemp = FormatTemplate("{name}_")
 
-    def visit_MaskBlock(self, node: npir.MaskBlock, **kwargs) -> Union[str, Collection[str]]:
+    def visit_MaskBlock(
+        self, node: npir.MaskBlock, *, is_serial: bool = False, **kwargs
+    ) -> Union[str, Collection[str]]:
         if isinstance(node.mask, npir.FieldSlice):
             mask_def = ""
         elif isinstance(node.mask, npir.BroadCast):
             mask_name = node.mask_name
             mask = self.visit(node.mask)
-            mask_def = f"{mask_name}_ = np.full((I - i, J - j, K - k), {mask})\n"
+            ksize = "1" if is_serial else "K - k"
+            mask_def = f"{mask_name}_ = np.full((I - i, J - j, {ksize}), {mask})\n"
         else:
             mask_name = node.mask_name
             mask = self.visit(node.mask)
             mask_def = f"{mask_name}_ = {mask}\n"
-        return self.generic_visit(node, mask_def=mask_def, **kwargs)
+        return self.generic_visit(node, mask_def=mask_def, is_serial=is_serial, **kwargs)
```

Comment:
If this is the only use of `is_serial` then why not go back to just passing it through via `kwargs`?